lisa/sut_orchestrator/aws/common.py

```diff
@@ -0,0 +1,207 @@
+from lisa.node import Node
+from lisa.util import LisaException, constants, field_metadata
+from lisa.util.parallel import check_cancelled
+from lisa.util.perf_timer import create_timer
+
+
+@dataclass
+class EnvironmentContext:
+    key_pair_name: str = ""
+    security_group_name: str = ""
+    security_group_id: str = ""
+    security_group_is_created: bool = False
+
+
+@dataclass
+class NodeContext:
+    intsance_id: str = ""
+    vm_name: str = ""
+    username: str = ""
+    password: str = ""
+    private_key_file: str = ""
+
+
+@dataclass_json()
+@dataclass
+class AwsVmMarketplaceSchema:
+    # Ubuntu Server 18.04 with SQL Server 2019 Express Edition AMI provided by Amazon
+    imageid: str = "ami-0340a222114f27094"
+
+
+@dataclass_json()
+@dataclass
+class AwsNodeSchema:
+    name: str = ""
+    vm_size: str = ""
+    location: str = ""
+
+    marketplace_raw: Optional[str] = field(
+        default=None, metadata=field_metadata(data_key="marketplace")
+    )
+
+    nic_count: int = 1
+    enable_sriov: bool = False
+    data_disk_count: int = 0
+    data_disk_caching_type: str = field(
+        default=constants.DATADISK_CACHING_TYPE_NONE,
+        metadata=field_metadata(
+            validate=validate.OneOf(
+                [
+                    constants.DATADISK_CACHING_TYPE_NONE,
+                    constants.DATADISK_CACHING_TYPE_READONLY,
+                    constants.DATADISK_CACHING_TYPE_READYWRITE,
+                ]
+            )
+        ),
+    )
+    data_disk_iops: int = 500
+    data_disk_size: int = 32
+    disk_type: str = ""
+
+    # for marketplace image, which need to accept terms
+    _marketplace: InitVar[Optional[AwsVmMarketplaceSchema]] = None
+
+    @property
+    def marketplace(self) -> Optional[AwsVmMarketplaceSchema]:
+        # this is a safe guard and prevent mypy error on typing
+        if not hasattr(self, "_marketplace"):
+            self._marketplace: Optional[AwsVmMarketplaceSchema] = None
+
+        if not self._marketplace:
+            assert isinstance(
+                self.marketplace_raw, str
+            ), f"actual: {type(self.marketplace_raw)}"
+            self.marketplace_raw = self.marketplace_raw.strip()
+            if self.marketplace_raw:
+                self._marketplace = AwsVmMarketplaceSchema(self.marketplace_raw)
+            else:
+                self._marketplace = AwsVmMarketplaceSchema()
+
+        return self._marketplace
+
+    @marketplace.setter
+    def marketplace(self, value: Optional[AwsVmMarketplaceSchema]) -> None:
+        self._marketplace = value
+
+    def get_image_id(self) -> str:
+        return self.marketplace.imageid
+
+
+def get_node_context(node: Node) -> NodeContext:
+    return node.get_context(NodeContext)
+
+
+def get_environment_context(environment: Environment) -> EnvironmentContext:
+    return environment.get_context(EnvironmentContext)
+
+
+def wait_operation(operation: Any, time_out: int = sys.maxsize) -> Any:
```

Comment:
This method is used to simplify the wait operation with timeout support. It should be used as many as possible.

Reply:
Remove this wait_operation function as it is not used. There is some wait method in AWS apis.